logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,135 @@
+ * be stored.
+ */
+public class AlluxioLogServerProcess implements LogServerProcess {
+  private ServerSocket mServerSocket;
+  private int mPort;
+  private String mBaseLogDir;
+  private HashMap<InetAddress, Hierarchy> mInetAddressHashMap;
+  private boolean mStopped;
+
+  public AlluxioLogServerProcess(String portStr, String baseLogDir) {
+    try {
+      mPort = Integer.parseInt(portStr);
+      mBaseLogDir = baseLogDir;
+      mServerSocket = new ServerSocket(mPort);
+      mInetAddressHashMap = new HashMap<>();
+      mStopped = false;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+  /**
+   * Starts the Alluxio process. This call blocks until the process is stopped via
+   * {@link #stop()}. The {@link #waitForReady()} method can be used to make sure that the
+   * process is ready to serve requests.
+   */
+  @Override
+  public void start() throws Exception {
+    while (!mStopped) {
+      Socket client = mServerSocket.accept();
+      InetAddress inetAddress = client.getInetAddress();
+      // Currently loggerRepository is always null because we are not using the map.
+      LoggerRepository loggerRepository = mInetAddressHashMap.get(inetAddress);
+      if (loggerRepository == null) {
+        loggerRepository = configureHierarchy(inetAddress);
+      }
+      try {
+        new Thread(new AlluxioLog4jSocketNode(client, loggerRepository)).start();
+      } catch (IOException e) {
+        // Could not create a thread to serve a client, ignore this client.
+        continue;
+      }
+    }
+  }
+
+  /**
+   * Stops the Alluxio process, blocking until the action is completed.
+   */
+  @Override
+  public void stop() throws Exception {
+    mStopped = true;
+  }
+
+  /**
+   * Waits until the process is ready to serve requests.
+   */
+  @Override
+  public void waitForReady() {
+    CommonUtils.waitFor(this + " to start", new Function<Void, Boolean>() {
+      @Override
+      public Boolean apply(Void input) {
+        return mServerSocket != null;
+      }
+    }, WaitForOptions.defaults().setTimeoutMs(10000));
+  }
+
+  private LoggerRepository configureHierarchy(InetAddress inetAddress) throws IOException, URISyntaxException {
+    Hierarchy clientHierarchy;
+    String inetAddressStr = inetAddress.toString();
+    int i = inetAddressStr.indexOf("/");
+    String key;
+    if (i == 0) {
+      key = inetAddressStr.substring(1, inetAddressStr.length());
+    } else {
+      key = inetAddressStr.substring(0, i);
+    }
+    Properties properties = new Properties();
+    File configFile = new File(new URI(System.getProperty("log4j.configuration")));
+    properties.load(new FileInputStream(configFile));
+    clientHierarchy = new Hierarchy(new RootLogger(Level.INFO));
+    String logFilePath = mBaseLogDir;
+    String loggerType = System.getProperty("alluxio.logger.type");
+    if (loggerType.contains("MASTER")) {
```

Comment:
Is `loggerType` normally going to be something like "REMOTE_MASTER_LOGGER" or "REMOTE_WORKER_LOGGER" here? If so, I'd prefer to log to a subdirectory named by the lowercase of the remote logger, e.g. `/remote_master_logger/ip_address.log`, `/remote_worker_logger/ip_address.log`, etc. That way this won't break if the name of the logger changes.

Reply:
Here the logger type will be MASTER_LOGGER or WORKER_LOGGER, because we need to configure the local appender on the log server. I feel we could re-use the properties in log4j.properties with a modified log file path.

Reply:
What if the directory does not exist? Do we create it in the ensure_dir function of the start script?

Reply:
The log server should create directories if they don't exist when it first receives a log event to the directory

Reply:
I'm a little doubtful about this because this behavior is different from normal local logging. The start script takes the responsibility of creating the directories if they do not exist. Should logserver have a different behavior?